### PR TITLE
Fix bluespace tomato and atom locking things (maybe)

### LINKED ISSUE #10884: Blue Space Tomatoes ; Handcuffs; And Buckled Chairs
#### Description of issue

Throwing a blue-space tomato at some one who is handcuffed and buckled in a chair, causes them to become stuck. Things like tazers, and other stuns, and pulling also don't effect the player in handcuffs. 
#### Difference between expected and actual behavior

After being hit by a bluespace tomatoe, and teleported while handcuffed in a chair you should be able to move. 
#### Steps to reproduce
1. Handcuff a player
2. Buckle them in a chair
3. Throw a bluespace tomatoe at them
4. Ensure they are teleported by the tomatoe
5. Check if they can move
#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [ ] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [ ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by gbasood**
Resolves #10884 probably
Some of this is actually overlapping but it's mostly a cherry-picked commit from https://github.com/Benblu/vgstation13/pull/41 by @Zaers 
One of his fixes assumed that `loc` is always `locked_to.loc`, and we have offsets for atom locking so I didn't think that'd be applicable
I can already see a bunch of stuff that'll probably need to be removed from this

### CODE REVIEW

File: code/game/objects/structures/vehicles/adminbus_powers.dm [hunk 1]
```diff
@@ -219,24 +219,18 @@
...
-			if(roadlights == 0)
-				overlays["roadlights"] = roadlights_image
 			roadlights = 1
 		if(2)
 			bususer.gui_icons.adminbus_roadlights_0.icon_state = "icon_lights_0-off"
 			bususer.gui_icons.adminbus_roadlights_1.icon_state = "icon_lights_1-off"
 			bususer.gui_icons.adminbus_roadlights_2.icon_state = "icon_lights_2-on"
 			lightsource.set_light(3)
-			if(roadlights == 0)
-				overlays["roadlights"] = roadlights_image
```
Comment (gbasood) [inline code/game/objects/structures/vehicles/adminbus_powers.dm:239]: I have zero idea why these changes are here

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -388,17 +388,21 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
...
-/mob/living/simple_animal/MouseDrop(mob/living/carbon/human/M)
-	if(M != usr || !istype(M) || !Adjacent(M) || M.incapacitated())
+/mob/living/simple_animal/MouseDrop(atom/A)
+	if(!Adjacent(A) )
 		return
 
 	if(locked_to) //Atom locking
 		return
 
-	var/strength_of_M = (M.size - 1) //Can only pick up mobs whose size is less or equal to this value. Normal human's size is 3, so his strength is 2 - he can pick up TINY and SMALL animals. Varediting human's size to 5 will allow him to pick up goliaths.
```
Comment (gbasood) [inline code/modules/mob/living/simple_animal/simple_animal.dm:398]: Or these

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 3]
```diff
@@ -86,17 +86,10 @@
 		target = M
 	if((target) && (target.op_stage.butt == 4)) //Butt surgery is at stage 4
 		if(!M.knockdown)	//Spam prevention
-			if(M == usr)
```
Comment (gbasood) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:89]: Wait is this actually a thing in our code still?

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 4]
```diff
@@ -99,7 +99,7 @@
 		to_chat(user, "<span class='warning'>The [M] is too squishy to buckle in.</span>")
 		return
 
-	if(M == usr)
```
Comment (gbasood) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:102]: WHAT

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm (hunk 3 above)
Comment (CptWad) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm]: i bet theres more in there

**Comment by gbasood** [comment]
actually I'm not sure this PR does what the title says at all

**Comment by PJB3005** [comment]
Nope it doesn't.

**Comment by Probe1** [comment]
iiam

**Comment by gbasood** [comment]
i mean some of this stuff should be redone in another PR because it cleans up shitcode
but i dont know what the fuck is going on anymore because i didnt cherry pick this
